### PR TITLE
Fix: Top SAM crashes

### PR DESCRIPTION
**Pull request by prenx4x**
*Issue #, if available:*

*Description of changes:*
Fixed some common SAM crashes

*Description of how you validated changes:*
Changes validated using unit tests

*Checklist:*

- [x] Write/update tests
- [x] `make pr` passes
- [ ] Update documentation
- [ ] Verify transformed template deploys and application functions as expected

*Examples?*

Please reach out in the comments, if you want to add an example. Examples will be 
added to `sam init` through https://github.com/awslabs/aws-sam-cli-app-templates/

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samtranslator/swagger/swagger.py [hunk 1]
```diff
@@ -621,6 +621,10 @@ def set_path_default_apikey_required(self, path):
                 # (e.g. sigv4 (AWS_IAM), authorizers, NONE (marker for ignoring default authorizer))
                 # We want to ensure only a single ApiKey security entry exists while keeping everything else
                 for security in existing_security:
+                    if not isinstance(security, dict):
+                        raise InvalidDocumentException(
+                            ["Invalid entry {security} in 'security'".format(security=security)]
```
Comment (jfuss) [inline samtranslator/swagger/swagger.py:626]: Why is it invalid?

File: samtranslator/swagger/swagger.py [hunk 2]
```diff
@@ -830,10 +834,10 @@ def add_models(self, models):
             model_properties = schema.get("properties")
 
             if not model_type:
-                raise ValueError("Invalid input. Value for type is required")
+                raise InvalidDocumentException(["Invalid input. Value for type is required"])
 
             if not model_properties:
-                raise ValueError("Invalid input. Value for properties is required")
+                raise InvalidDocumentException(["Invalid input. Value for properties is required"])
```
Comment (jfuss) [inline samtranslator/swagger/swagger.py:840]: Same here

File: samtranslator/swagger/swagger.py [hunk 3]
```diff
@@ -846,6 +850,8 @@ def add_resource_policy(self, resource_policy, path, api_id, stage):
         """
         if resource_policy is None:
             return
+        if not isinstance(resource_policy, dict):
+            raise InvalidDocumentException(["Invalid resource policy"])
```
Comment (jfuss) [inline samtranslator/swagger/swagger.py:854]: Make sure to provide some details about where this is in the template. We generally do this through adding the logical id, so the customer at least knows what resource they need to dig into. This goes for all the exceptions, not just this one.

**Review by jfuss — CHANGES_REQUESTED** [review]
Can we add additional test cases in `test_transform_invalid_document` to verify we have fixed the issue.

**Comment by prenx4x** [comment]
We can close this PR as I am breaking it in favor of #1757 and others to come.

**Pull request closed without merging**